### ISSUE TITLE
`fopen` returns seemingly valid pointer if file not found

### ISSUE THREAD

**Issue by giulianobelinassi** [DragonMinded/libdragon#149]
With the following patch for spritemap.c:
```
diff --git a/examples/spritemap/spritemap.c b/examples/spritemap/spritemap.c
index 1174e52..46047f5 100644
--- a/examples/spritemap/spritemap.c
+++ b/examples/spritemap/spritemap.c
@@ -24,13 +24,19 @@ int main(void)
     rdp_init();
     controller_init();
     timer_init();
+    debug_init_isviewer();
 
     /* Read in single sprite */
     int fp = dfs_open("/mudkip.sprite");
     sprite_t *mudkip = malloc( dfs_size( fp ) );
     dfs_read( mudkip, 1, dfs_size( fp ), fp );
     dfs_close( fp );
-    
+
+    FILE* fptr = fopen("rom://should_not_exist", "r");
+    if (!fptr)
+        debugf("File not found\n");
+    fclose(fptr);
+
     fp = dfs_open("/earthbound.sprite");
     sprite_t *earthbound = malloc( dfs_size( fp ) );
     dfs_read( earthbound, 1, dfs_size( fp ), fp );
```
Clearly, `should_not_exist` is not present in the DragonFS. However, running with:
```
cen64 -is-viewer ~/bagunca/n64pif/pifdata.bin spritemap.z64
```
The __file not found__ message is never print, and a seemly valid (but invalid) pointer is returned, where it should have returned `NULL`.

**Comment by rasky** [comment]
I think I've fixed this bug here:

https://github.com/DragonMinded/libdragon/commit/86c824cf13f3948843a11c2af7bbed4509a99ce4

Can you please update and try again?

**Comment by giulianobelinassi** [comment]
Indeed. this was fixed by 86c824c.
